src/workers/algorithm/index.ts

```diff
@@ -0,0 +1,28 @@
+import { AlgorithmResult } from '../../algorithms/types/Result.types'
+import { MessageData } from './worker.types'
+
+export async function run(...args) {
+  return new Promise<AlgorithmResult>((resolve, reject) => {
+    const worker = new Worker('./worker.ts', { type: 'module' })
+
+    worker.addEventListener('message', (message: MessageEvent) => {
+      const { result, error }: MessageData = message.data
+
+      if (result) {
+        resolve(result)
+        return
+      }
+
+      if (error) {
+        reject(error)
+      }
+    })
+
+    worker.addEventListener('error', (error) => {
+      reject(error)
+      console.error('Worker error:', error)
+    })
+
+    worker.postMessage(args)
+  })
+}
```

Comment:
Wraps webworker interaction into a promise suitable for `await`. Initialize the worker, add a message handler, start the algorithm, and `resolve` the result. Algorithm exceptions and worker errors are propagated via `reject`.